costs/es_request_constructor_test.go

```diff
@@ -409,3 +410,324 @@ func TestAggregationNestingWithAllHandledElasticAggregationTypes(t *testing.T) {
+	}
+}
+
+func TestElasticSearchParamWithFewResultsAndNoAggregationNesting(t *testing.T) {
+	client, err := elastic.NewClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	accountList := []string{"394125495069"}
+	durationBegin, _ := time.Parse("2006-1-2 15:04", "2017-01-12 11:23")
+	durationEnd, _ := time.Parse("2006-1-2 15:04", "2017-05-23 11:23")
+	params := []string{"product"}
+	index := "awsdetailedlineitem"
+	expectedResult := `{
+	"product": {
+		"doc_count_error_upper_bound": 0,
+		"sum_other_doc_count": 0,
+		"buckets": [
+			{
+				"key": "Amazon Elastic Compute Cloud",
+				"doc_count": 49,
+				"cost": {
+					"value": 0.14448613000000002
+				}
+			},
+			{
+				"key": "Amazon DynamoDB",
+				"doc_count": 18,
+				"cost": {
+					"value": 2.0E-8
+				}
+			},
+			{
+				"key": "Amazon Simple Storage Service",
+				"doc_count": 13,
+				"cost": {
+					"value": 2.4008000000000002E-4
+				}
+			},
+			{
+				"key": "AWS Lambda",
+				"doc_count": 7,
+				"cost": {
+					"value": 2.34E-6
+				}
+			},
+			{
+				"key": "Amazon Elasticsearch Service",
+				"doc_count": 3,
+				"cost": {
+					"value": 0.1993281
+				}
+			},
+			{
+				"key": "Amazon RDS Service",
+				"doc_count": 3,
+				"cost": {
+					"value": 0.0016027799999999998
+				}
+			},
+			{
+				"key": "Amazon Simple Notification Service",
+				"doc_count": 3,
+				"cost": {
+					"value": 0.0
+				}
+			},
+			{
+				"key": "Amazon EC2 Container Registry (ECR)",
+				"doc_count": 1,
+				"cost": {
+					"value": 2.765E-5
+				}
+			},
+			{
+				"key": "Amazon Route 53",
+				"doc_count": 1,
+				"cost": {
+					"value": 3.6E-6
+				}
+			},
+			{
+				"key": "Amazon Simple Queue Service",
+				"doc_count": 1,
+				"cost": {
+					"value": 0.0
+				}
+			},
+			{
+				"key": "AmazonCloudWatch",
+				"doc_count": 1,
+				"cost": {
+					"value": 0.0
+				}
+			}
+		]
+	}
+}`
+	searchService := GetElasticSearchParams(accountList, durationBegin, durationEnd, params, client, index)
+	res, err := searchService.Do(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	aggregationMarshalled, err := json.MarshalIndent(res.Aggregations, "", "	")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(aggregationMarshalled) != expectedResult {
+		t.Fatalf("Expected %v but got %v", expectedResult, string(aggregationMarshalled))
+	}
+}
+
+func TestElasticSearchParamWithFewResultsAndNesting(t *testing.T) {
+	client, err := elastic.NewClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	accountList := []string{"394125495069"}
+	durationBegin, _ := time.Parse("2006-1-2 15:04", "2017-01-12 11:23")
+	durationEnd, _ := time.Parse("2006-1-2 15:04", "2017-05-23 11:23")
+	params := []string{"product", "region", "day"}
+	index := "awsdetailedlineitem"
+	expectedResult := `{
+	"product": {
+		"doc_count_error_upper_bound": 0,
+		"sum_other_doc_count": 0,
+		"buckets": [
+			{
+				"key": "Amazon Elastic Compute Cloud",
```

Comment:
Is this order of buckets guaranteed ?

Reply:
I don't know if the order will be preserved.

I have also found out that the Elasticsearch will have some rounding errors : 

```
< 							"value": 2.4008000000000002E-4
---
> 							"value": 2.4008E-4
```

I will have to redo the way I validate the output of the Elasticsearch